### DISCUSSION TITLE
how do I add Modern Fortran to my custom IDE?

### DISCUSSION THREAD

**Discussion by peytonk132** [krvajal/vscode-fortran-support#1092] [General]
I don't know if this question is asked often. I've made an IDE for Fortran for fun to try and give me a big project to work on. I used ImGui and SFML to make it but now I want to add Modern Fortran for systax highlighting, code completion, etc. same with FortLS. All the tutorials I've found online is everybody doing it for Vim or VSCode but those, I feel, can't really help me. I'm still new to most of this so if you could help me out, that'd be great!

**Comment by gnikit** [comment]
Modern Fortran (i.e. this extension) is probably hard to add in anything other than a VS Code-based editor. The extension is designed using the plugin API of vscode, so there are too many dependencies and vscode specific syntax to be portable in other editors.

If it's okay with you I will respond to you in our Fortran-lang Discourse thread to keep the discussion in one place.
